jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCPoolManager.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.zeppelin.jdbc;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.InterpreterResultMessage;
+import org.apache.zeppelin.resource.ResourcePool;
+import org.apache.zeppelin.resource.WellKnownResourceName;
+import org.apache.zeppelin.tabledata.ColumnDef;
+import org.apache.zeppelin.tabledata.InterpreterResultTableData;
+import org.apache.zeppelin.tabledata.Row;
+import org.apache.zeppelin.tabledata.TableData;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class JDBCPoolManager {
+  private static final String PARAGRAPH_ID_KEY = "paragraph_id=";
+  private static final String NOTE_ID_KEY = "note_id=";
+  // the table name must begin with a letter
+  private static final String SQL_NAME_PREFIX = "p";
+
+  private static String getPoolExpression(String sqlReq) {
+    final int poolReqBegIndex = sqlReq.indexOf("{ResourcePool");
```

Comment:
this isn't very safe - the sql query could contain the same text

Reply:
Common SQL syntax almost not contain figure brackets. Especially with word "ResourcePool" in start of it.

Reply:
what I'm saying is we could be more strict/careful in parsing the sql, not just indexOf

Reply:
Fixed